platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/metrics/ReconnectMetrics.java

```diff
@@ -54,12 +68,27 @@ public class ReconnectMetrics {
      * 		reference to the metrics-system
      * @throws IllegalArgumentException if {@code metrics} is {@code null}
      */
-    public ReconnectMetrics(final Metrics metrics) {
-        CommonUtils.throwArgNull(metrics, "metrics");
+    public ReconnectMetrics(@NonNull final Metrics metrics, @NonNull final AddressBook addressBook) {
+        Objects.requireNonNull(metrics, "metrics");
+        Objects.requireNonNull(addressBook, "addressBook");
         senderStartTimes = metrics.getOrCreate(SENDER_START_TIMES_CONFIG);
         receiverStartTimes = metrics.getOrCreate(RECEIVER_START_TIMES_CONFIG);
         senderEndTimes = metrics.getOrCreate(SENDER_END_TIMES_CONFIG);
         receiverEndTimes = metrics.getOrCreate(RECEIVER_END_TIMES_CONFIG);
+
+        for (final Address address : addressBook) {
+            final NodeId nodeId = address.getNodeId();
+            rejectionFrequency.put(
+                    nodeId,
+                    new CountPerSecond(
+                            metrics,
+                            new CountPerSecond.Config(
```

Comment:
I believe this metric should be given a unit, though beware that the string `\s` is forbidden in unit names, due to prometheus quirks

Reply:
adding `withUnits("rejectionsPerSec")`